src/main.rs

```diff
@@ -1,4 +1,146 @@
 // This is here to make Cargo happy, for now
-fn main() {
-    println!("Hello, world!");
+use ethers::utils::{Anvil};
+use ethers_contract::{abigen, Contract};
+use ethers_middleware::signer::SignerMiddleware;
+use ethers_providers::{Provider, Http, Middleware};
+use ethers::signers::{Signer, Wallet};
+use ethers::prelude::LocalWallet;
+use ethers_core::types::{Address, H160};
+use std::sync::Arc;
+use std::time::Duration;
+use async_std::task;
+use ethers::abi::Uint;
+use std::env;
+
+// Generate Rust bindings for all required contracts
+abigen!(SevenTeenTwentyNineEssay, "out/1729Essay.sol/SevenTeenTwentyNineEssay.json");
+abigen!(ListBidEssayScript, "out/ListBidEssay.s.sol/ListBidEssayScript.json");
+abigen!(ReserveAuctionCoreETH, "out/ListBidEssay.s.sol/ReserveAuctionCoreETH.json");
+abigen!(ModuleManager, "out/ListBidEssay.s.sol/ModuleManager.json");
+
+
+
+#[tokio::main]
+async fn main() {
+    let fork_endpoint = env::var("ETH_NODE_URL").expect("Environment variable ETH_NODE_URL should be defined and be a valid API URL");
+    let anvil = Anvil::new()
+        .fork(fork_endpoint)
+        .spawn();
+    let endpoint = anvil.endpoint();
+    println!("Anvil running at `{}`", endpoint);
+
+    // Set up anvil
+    let provider = Provider::<Http>::try_from(anvil.endpoint()).expect("Failed to connect to Anvil").interval(Duration::from_millis(10u64));
```

Comment:
Not blocking, wondering if we'll reframe expect messages to form of "ABC should XYZ".

Reply:
We should.  I was thinking of doing that, and some other refactoring, when converting this to stepdefs.